code/Player/PropPossession.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System;
+using System.Linq;
+using Sandbox;
+using TTT.UI;
+
+namespace TTT;
+
+public partial class PropPossession : EntityComponent<Player>
+{
+	public const int MaxPunches = 8;
+	private const float RechargeTime = 1f;
+
+	[Net, Local]
+	public int Punches { get; private set; }
+
+	[Net]
+	private Prop Prop { get; set; }
+
+	private Player _player;
+	private TimeUntil _timeUntilRecharge = 0;
+	private TimeUntil _timeUntilNextPunchAllowed = 0;
+
+	// Clientside
+	private PossessionNameplate _nameplate;
+	private PunchMeter _meter;
+
+	public PropPossession() { }
+
+	public PropPossession( Prop prop ) => Prop = prop;
+
+	protected override void OnActivate()
+	{
+		_player = Entity;
+
+		if ( Host.IsServer )
+		{
+			Prop.Owner = _player;
+			_player.Camera = new FollowEntityCamera( Prop );
+			_timeUntilRecharge = RechargeTime;
+		}
+		else
+		{
+			if ( !_player.IsAlive() )
+				_nameplate = new PossessionNameplate( _player, Prop );
```

Comment:
I believe `_player` will always be not alive because only not alive players are able to possess props.
Since `OnActivate()` is called on every client, this would make possession nameplates visible for every player (including alive players) if im not misunderstanding the logic here.